File: android/src/main/kotlin/no/nordicsemi/android/mcumgr_flutter/utils/StreamHandler.kt
package no.nordicsemi.android.mcumgr_flutter.utils

import io.flutter.plugin.common.EventChannel

class StreamHandler: EventChannel.StreamHandler {
	var sink: EventChannel.EventSink? = null
		private set

	override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
		sink = events
	}

	override fun onCancel(arguments: Any?) {
		sink = null
	}
}

File: darwin/Classes/Utils/StreamHandler.swift
//
//  StreamHandler.swift
//  mcumgr_flutter
//
//  Created by Mykola Kibysh on 13/12/2020.
//

#if os(iOS)
import Flutter
import UIKit
#elseif os(macOS)
import FlutterMacOS
import AppKit
#endif

class StreamHandler: NSObject, FlutterStreamHandler {
    private(set) var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.sink = nil
        return nil
    }
    
    
}
